src/coreclr/System.Private.CoreLib/src/Microsoft/Win32/OAVariantLib.cs

```diff
@@ -32,96 +36,175 @@ internal static unsafe partial class OAVariantLib
-            typeof(float),
-            typeof(double),
-            typeof(string),
-            typeof(void),
-            typeof(DateTime),
-            typeof(TimeSpan),
-            typeof(object),
-            typeof(decimal),
-            null,  // Enums - what do we do here?
-            typeof(Missing),
-            typeof(DBNull),
+        private static readonly Dictionary<Type, VarEnum> ClassTypes = new Dictionary<Type, VarEnum>
+        {
+            { typeof(Empty), VarEnum.VT_EMPTY },
+            { typeof(void), VarEnum.VT_VOID },
+            { typeof(bool), VarEnum.VT_BOOL },
+            { typeof(char), VarEnum.VT_I2 },
+            { typeof(sbyte), VarEnum.VT_I1 },
+            { typeof(byte), VarEnum.VT_UI1 },
+            { typeof(short), VarEnum.VT_I2 },
+            { typeof(ushort), VarEnum.VT_UI2 },
+            { typeof(int), VarEnum.VT_I4 },
+            { typeof(uint), VarEnum.VT_UI4 },
+            { typeof(long), VarEnum.VT_I8 },
+            { typeof(ulong), VarEnum.VT_UI8 },
+            { typeof(float), VarEnum.VT_R4 },
+            { typeof(double), VarEnum.VT_R8 },
+            { typeof(string), VarEnum.VT_BSTR },
+            { typeof(DateTime), VarEnum.VT_DATE },
+            { typeof(object), VarEnum.VT_UNKNOWN },
+            { typeof(decimal), VarEnum.VT_DECIMAL },
         };
 
-        // Keep these numbers in sync w/ the above array.
-        private const int CV_OBJECT = 0x12;
-
         #endregion
 
 
         #region Internal Methods
 
-#pragma warning disable CS8500
-
         /**
          * Changes a Variant from one type to another, calling the OLE
          * Automation VariantChangeTypeEx routine.  Note the legal types here are
          * restricted to the subset of what can be legally found in a VB
          * Variant and the types that CLR supports explicitly in the
          * CLR Variant class.
          */
-        internal static Variant ChangeType(Variant source, Type targetClass, short options, CultureInfo culture)
+        internal static object? ChangeType(object source, Type targetClass, short options, CultureInfo culture)
         {
             ArgumentNullException.ThrowIfNull(targetClass);
             ArgumentNullException.ThrowIfNull(culture);
 
-            Variant result = default;
-            ChangeType(
-                &result,
-                &source,
-                culture.LCID,
-                targetClass.TypeHandle.Value,
-                GetCVTypeFromClass(targetClass),
-                options);
-            return result;
-        }
+            object? result = null;
 
-        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "OAVariant_ChangeType")]
-        private static partial void ChangeType(Variant* result, Variant* source, int lcid, IntPtr typeHandle, int cvType, short flags);
+            if (Variant.IsSystemDrawingColor(targetClass))
+            {
+                if (source.GetType() == typeof(int) || source.GetType() == typeof(uint))
```

Comment:
The `if (source is int i || source is uint j)` pattern is preferred in this case.

Reply:
Unfortunately, pattern variables aren't definitely assigned under OR cases, unlike AND.

Reply:
Okay, then we can drop the assignment component and just perform the checks.

Reply:
```suggestion
                if (source is int || source is uint)
```

Reply:
I'm a bit resistant to `is` because of #47920. Not a real issue though.

Reply:
I prefer the `is` checks. Those patterns are what C# recommends and thus what we will, in the long term, focus on optimizing.